crates/matrix-sdk/src/send_queue.rs

```diff
@@ -577,6 +589,28 @@ impl RoomSendQueue {
     }
 }
 
+impl From<&crate::Error> for QueueWedgeError {
```

Comment:
nit: using a `From` impl makes it part of the public API of `QueueWedgeError`, which we probably don't want to do, since it's an internal conversion. Can you use a plain function instead, please?

Also, take the `crate::Error`  by ownership (i.e. no ref `&`), since this could avoid some cloning if I'm reading the code correctly.

Reply:
It is used in the timeline controller to update the in memory local event EventSendState. Why is it a issue that it is a public API?